File: java/Clothing.java
public class Clothing extends Product {
    private String size;
    private String material;
    private String gender;

    public Clothing(){
    }

    public void setSize(String size){
        this.size = size;
    }

    public String getSize(){
        return this.size;
    }

    public void setMaterial(String material){
        this.material = material;
    }

    public String getMaterial(){
        return this.material;
    }

    public void setGender(String gender){
        this.gender = gender;
    }

    public String getGender(){
        return this.gender;
    }
}

File: cpp/Clothing.cpp
#include <iostream>
#include <string>
#include "Product.cpp"

using namespace std;

class Clothing : public Product
{
    private:
        string size;
        string material;
        string gender;

    public:
        Clothing(){
        }

        void setSize(string size){
            this->size = size;
        }

        string getSize(){
            return this->size;
        }
        
        void setMaterial(string material){
            this->material = material;
        }

        string getMaterial(){
            return this->material;
        }

        void setGender(string gender){
            this->gender = gender;
        }

        string getGender(){
            return this->gender;
        }

        ~Clothing(){
        }
};